arbitrum-client/src/conversion.rs

```diff
@@ -141,6 +148,43 @@ pub fn to_contract_valid_wallet_update_statement(
     })
 }
 
+/// Convert a [`TransferAuxData`] to its corresponding smart contract type
+pub fn to_contract_transfer_aux_data(
```

Comment:
I'd suggest having this method take in `TransferAuxData`, implement `Default` on `ContractTransferAuxData` and in the caller we do a `.map(<this-method>).unwrap_or_default()`

This method's control flow would be cleaned up